### PR TITLE
Add autotools example using AutotoolsToolchain with libfmt

### LINKED ISSUE #3064: Create your first Conan package with Autotools example
An example [for the examples section](https://docs.conan.io/2/examples.html), similar to this part of the tutorial https://docs.conan.io/2/tutorial/creating_packages/create_your_first_package.html but for Autotools.

### PR DESCRIPTION
**Pull request by uilianries**
An example based on Meson example, but now using C++ and fmt project instead zlib.

For Windows it would be more complicated to reproduce, as it would mingw or similar to use gnu tools like automake.

**requires https://github.com/conan-io/examples2/pull/84 first**

closes #3064

Current preview:

Screenshot 2023-03-30 at 13-39-08 Build a simple Autotools project using Conan — conan 2 0 2 documentation

### CODE REVIEW

**Comment by czoido** [comment]
Great! Please, target the release/2.0 branch as this is not related to the release and could be published when merged

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+Build a simple Autotools project using Conan
+============================================
+
+In this example, we are going to create a string formatter application
+that uses one of the most popular C++ libraries: `fmt <https://fmt.dev/latest/index.html/>`__.
+
+We'll use Autotools as build system and pkg-config as helper tool in this case, so you should get them installed
+on Linux and Mac before going forward with this example. In case you do not have them installed already, you can
+`Conan Center <https://conan.io/center/>`_ and install them as build requirements.
```
Comment (czoido) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:11]: I would remove the sentence about installing them as build requirements, because if this tutorial is example is for newbies it may be a bit confusing, the most typical thing is to have that installed on the system.

Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Done.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+    #include <cstdlib>
+    #include <fmt/core.h>
+
+    int main(int argc, char** argv) {
+        fmt::print("{} - The C++ Package Manager!\n", "Conan");
+        return EXIT_SUCCESS;
+    }
+
+The ``configure.ac`` tells us about our project will a C++ compiler and a pkg-config module named as ``fmt.pc``
```
Comment (czoido) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:47]: This sentence is a bit confusing, can we rephrase this? Something like this?

The ``configure.ac`` file checks for a C++ compiler using the ``AC_PROG_CXX`` macro and also checks for the `fmt.pc` pkg-config module using the PKG_CHECK_MODULES macro.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    AC_INIT([stringformatter], [0.1.0])
+    AM_INIT_AUTOMAKE([1.10 -Wall no-define foreign])
+    AC_CONFIG_SRCDIR([src/main.cpp])
+    AC_CONFIG_FILES([Makefile])
+    PKG_CHECK_MODULES([fmt], [fmt])
+    AC_PROG_CXX
+    AC_OUTPUT
+
+And, the *Makefile.am* has ``string_formatter`` as our expected executable, and it will linked to the ``fmt`` library.
+
```
Comment (czoido) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:61]: Maybe rephrase this?

```suggestion
The *Makefile.am* specifies that ``stringformatter`` is the expected executable and that it should be linked to the ``fmt`` library.

```

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    AUTOMAKE_OPTIONS = subdir-objects
+    ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS}
+
+    bin_PROGRAMS = string_formatter
+    string_formatter_SOURCES = src/main.cpp
+    string_formatter_CPPFLAGS = $(fmt_CFLAGS)
+    string_formatter_LDADD = $(fmt_LIBS)
+
+
+The *conanfile.txt* looks simple, just install **fmt** and uses two generatos to build our project
```
Comment (czoido) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:74]:
```suggestion
The *conanfile.txt* looks simple as it just installs the **fmt** package and uses two generators to build our project.```

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+to *autotools* using a `conanbuild[.sh|.bat]` file that describes the compilation environment.
+
+We will use Conan to install **fmt** and generate the files that Autotools needs to find this library and build our project.
+
+
+Building on Linux and macOS
+---------------------------
+
+First, we should install the requirements. For this example we didn't consider your profile, but ``fmt`` is higly compatible with many different configurations.
+
```
Comment (czoido) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:97]: Maybe we can add more context to the sentence: "For this example we didn't consider your profile"

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    conan install . --build=missing
+
+Now we are ready to build and run our **string_formatter** app:
+
+.. code-block:: shell
+
+    source conanbuild.sh
+
+The ``conanbuild.sh`` generated by the :ref:`VirtualBuildEnv<conan_tools_env_virtualbuildenv>` will fill all environment
+variables according to your profile, plus, Autotools, fmt .pc files and ``PKG_CONFIG_PATH`` as environment variable.
```
Comment (czoido) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:109]: I think it would be nice to rephrase this and explain a bit more of what each AutotoolsToolchain and PkgConfigDeps generators did after the install.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 7]
```diff
@@ -0,0 +1,214 @@
+.. _examples_tools_autotools_autotools_toolchain_build_project_autotools_toolchain:
+
+Build a simple Autotools project using Conan
+============================================
+
+In this example, we are going to create a string formatter application
+that uses one of the most popular C++ libraries: `fmt <https://fmt.dev/latest/index.html/>`__.
+
+We'll use Autotools as build system and pkg-config as helper tool in this case, so you should get them installed
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:9]:
```suggestion
We'll use `Autotools <https://www.gnu.org/software/automake/manual/html_node/Autotools-Introduction.html>`_ as build system and `pkg-config <https://www.freedesktop.org/wiki/Software/pkg-config/>`_ as helper tool in this case, so you should get them installed
```
a reader might not be familiar what are these things, providing some links would be helpful

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+on Linux and Mac before going forward with this example.
+
+Please, at first, clone the sources to recreate this project, you can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ in GitHub:
+
+.. code-block:: shell
+
+    git clone https://github.com/conan-io/examples2.git
+    cd examples2/examples/tools/autotools/autotoolstoolchain/string_formatter
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:18]: I get no such file or directory

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+Let's have a look at the *main.cpp* file, it only prints a default message, but uses ``fmt::print`` method for it.
+
+.. code-block:: cpp
+    :caption: **main.c**
+
+    #include <cstdlib>
+    #include <fmt/core.h>
+
+    int main(int argc, char** argv) {
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:42]:
```suggestion
    int main() {
```
not used there, may trigger a warning (unused argument)

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+This extras steps are only needed because we are building our project manually on the terminal,
+but in case it would be a regular Conan recipe, building a Conan package, it would not be necessary.
+Now we are ready to build the example.
+
+Now we should install and build the ``fmt`` with the msys custom profiles.
+Note that we do not need to use activated bash configuration for this action, otherwise, it may raise an unix path error when building the library.
+
+.. code-block:: shell
+
+    conan install -r conancenter . --build=missing --profile:host=profiles/msys -c tools.microsoft.bash:active=False
```
Comment (memsharded) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:190]: This is a bit confusing to me, this shouldn't be the case.
If there is a gap in ``fmt`` recipe for ``msys2``, then probably better address it first.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+
+    AC_INIT([stringformatter], [0.1.0])
+    AM_INIT_AUTOMAKE([1.10 -Wall no-define foreign])
+    AC_CONFIG_SRCDIR([src/main.cpp])
+    AC_CONFIG_FILES([Makefile])
+    PKG_CHECK_MODULES([fmt], [fmt])
+    AC_PROG_CXX
+    AC_OUTPUT
+
+And, the *Makefile.am* has ``string_formatter`` as our expected executable, and it will linked to the ``fmt`` library.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:60]:
```suggestion
And, the *Makefile.am* has ``string_formatter`` as our expected executable, and it will be linked to the ``fmt`` library.
```
passive voice?

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+    [requires]
+    fmt/9.1.0
+
+    [generators]
+    AutotoolsToolchain
+    PkgConfigDeps
+
+In this case, we will use :ref:`PkgConfigDeps<conan_tools_gnu_pkgconfigdeps>` to generate information about where the **fmt** library
+files are installed thanks to the `*.pc` files and :ref:`AutotoolsToolchain<conan_tools_gnu_autotoolstoolchain>` to pass build information
+to *autotools* using a `conanbuild[.sh|.bat]` file that describes the compilation environment.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:88]: need some link to reference, what is this `conanbuild` file

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 8 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:18]: It requires https://github.com/conan-io/examples2/pull/84 first

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+
+We will use Conan to install **fmt** and generate the files that Autotools needs to find this library and build our project.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:90]: what are the files to be generated?

**Review by memsharded — COMMENTED** [review]
I still have some concerns about the Windows part, like if we should target ``msvc`` compiler first, and not ``gcc`` (as all recipes in ConanCenter are tested for ``msvc``), the need of many ``tool_requires`` (not clear for users that already run in their own msys2), or the fact that ``fmt`` might be failing if we don't set the active=False, which is confusing.

I'd suggest separating the Windows part, so we can merge asap the Linux/OSX docs, which is very good to have it asap.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+
+
+Building on Linux and macOS
+---------------------------
+
+First, we should install the requirements. For this example we didn't consider your profile, but ``fmt`` is higly compatible with many different configurations.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:96]: I am not sure who are `we` in this context. maybe use unspecified sentence like  `your profile wasn't considered`

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+
+.. code-block:: shell
+
+    aclocal
+    automake --add-missing
+    autoconf
+    ./configure
+    make
+
+Now we generate the ``configure`` and ``Makefile``, then we build our project by running the ``make`` command.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:119]: it would be better to explain each step a little bit more here

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+
+As final result, we will have the binary ``string_formatter`` installed in our example folder.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:121]: don't we already have it by that time?

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+    [tool_requires]
+    msys2/cci.latest
+    autoconf/2.71
+    automake/1.16.5
+    pkgconf/1.9.3
+    [buildenv]
+    CC=gcc
+    CXX=g++
+
+With this profile, we will build on Windows, with MSYS2 support and GCC 11. Some build requirements will be needed too.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:162]:
```suggestion
With this profile, we will build on Windows, with MSYS2 support and GCC 11. Some tool requirements will be needed too.
```

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+
+As the first step, we will install MSYS2 Conan package, to be able to run bash inside of our command prompt.
+
+.. code-block:: batch
+
+    conan install -r conancenter --requires=msys2/cci.latest -g VirtualBuildEnv
+    conanbuild.bat
+    set MSYSTEM=MSYS
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:170]: why not put it into profile?

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 19]
```diff
@@ -0,0 +1,214 @@
...
+    set MSYS2_PATH_TYPE=inherit
+    set CHERE_INVOKING=1
+    %MSYS_BIN%\bash.exe -l
+
+Once we have installed ``msys2/cci.latest`` package, we can activate the environment generated by :ref:`VirtualBuildEnv<conan_tools_env_virtualbuildenv>`.
+Then, we need to configure some environment variables required by MSYS2.
+
+The environment variable ``%MSYS_BIN%`` is configured by the script ``conanbuild.bat`` so we do not need to find where the Conan package is installed.
+As next step, we start the bash terminal embedded in the Conan package.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:179]: a terminal is not embedded, I don't think this sentence is technically correct, and it's rather confusing. terminal emulator is embedded (which is mintty), bash is not a terminal, it's sell, and terminal (tty) will be created once bash is launched.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 20]
```diff
@@ -0,0 +1,214 @@
...
+
+This extras steps are only needed because we are building our project manually on the terminal,
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:181]:
```suggestion
These extra steps are only needed because we are building our project manually on the terminal,
```

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 21]
```diff
@@ -0,0 +1,214 @@
...
+but in case it would be a regular Conan recipe, building a Conan package, it would not be necessary.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:182]: what is `it` here? (both cases)

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 22]
```diff
@@ -0,0 +1,214 @@
...
+Now we are ready to build the example.
+
+Now we should install and build the ``fmt`` with the msys custom profiles.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:185]:
```suggestion
Now we should install and build the ``fmt`` with the MSYS custom profiles.
```

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 23]
```diff
@@ -0,0 +1,214 @@
...
+Note that we do not need to use activated bash configuration for this action, otherwise, it may raise an unix path error when building the library.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:186]: what is `bash configuration` in this context?

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 24]
```diff
@@ -0,0 +1,214 @@
...
+
+.. code-block:: shell
+
+    conan install -r conancenter . --build=missing --profile:host=profiles/msys -c tools.microsoft.bash:active=False
+
+And now we can generate shell scripts that should be loaded by our terminal and will configure our settings, options, and environment variables like ``PKG_CONFIG_PATH``.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:192]: they will be loaded by shell, not a terminal

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 25]
```diff
@@ -0,0 +1,214 @@
...
+    autoconf/2.71
+    automake/1.16.5
+    pkgconf/1.9.3
+    [buildenv]
+    CC=gcc
+    CXX=g++
+
+With this profile, we will build on Windows, with MSYS2 support and GCC 11. Some build requirements will be needed too.
+
+As the first step, we will install MSYS2 Conan package, to be able to run bash inside of our command prompt.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:164]:
```suggestion
As the first step, we will install `MSYS2 <https://www.msys2.org/>`_ Conan package, to be able to run bash inside of our command prompt.
```

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 26]
```diff
@@ -0,0 +1,214 @@
...
+
+    conan install -r conancenter --requires=msys2/cci.latest -g VirtualBuildEnv
+    conanbuild.bat
+    set MSYSTEM=MSYS
+    set MSYS2_PATH_TYPE=inherit
+    set CHERE_INVOKING=1
+    %MSYS_BIN%\bash.exe -l
+
+Once we have installed ``msys2/cci.latest`` package, we can activate the environment generated by :ref:`VirtualBuildEnv<conan_tools_env_virtualbuildenv>`.
+Then, we need to configure some environment variables required by MSYS2.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:176]:
```suggestion
Then, we need to `configure <https://www.msys2.org/wiki/Launchers/>`_ some `environment variables <https://www.msys2.org/wiki/MSYS2-introduction/>`_ required by MSYS2.
```
it's a bit unclear what are these variables we're setting here

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 27]
```diff
@@ -0,0 +1,214 @@
...
+    └── src
+        └── main.cpp
+
+This project contains a basic `configure.ac <https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Writing-configure_002eac.html>_` including the **fmt** pkg-config dependency and the
+source code for the string formatter program in *main.cpp*.
+
+Let's have a look at the *main.cpp* file, it only prints a default message, but uses ``fmt::print`` method for it.
+
+.. code-block:: cpp
+    :caption: **main.c**
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:37]:
```suggestion
    :caption: **main.cpp**
```
it won't compile as pure c

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 28]
```diff
@@ -0,0 +1,214 @@
...
+    ├── configure.ac
+    ├── Makefile.am
+    ├── conanfile.txt
+    └── src
+        └── main.cpp
+
+This project contains a basic `configure.ac <https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Writing-configure_002eac.html>_` including the **fmt** pkg-config dependency and the
+source code for the string formatter program in *main.cpp*.
+
+Let's have a look at the *main.cpp* file, it only prints a default message, but uses ``fmt::print`` method for it.
```
Comment (SSE4) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:34]:
```suggestion
Let's have a look at the *main.cpp* file, it only prints a simple message, but uses ``fmt::print`` method for it.
```
I don't thing `default` is right word here

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 2 above)
Comment (SSE4) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:47]: yes, please change. it doesn't `tell`, as it's not living entity, I think it's confusing for non-native speakers to use metaphors.

**Comment by uilianries** [comment]
> I still have some concerns about the Windows part, like if we should target `msvc` compiler first, and not `gcc` (as all recipes in ConanCenter are tested for `msvc`), the need of many `tool_requires` (not clear for users that already run in their own msys2), or the fact that `fmt` might be failing if we don't set the active=False, which is confusing.
> 
> I'd suggest separating the Windows part, so we can merge asap the Linux/OSX docs, which is very good to have it asap.

Yes, it makes sense. I just removed the Windows section now, so let's explore NMake + msvc scenario later. Like a said before, Autotools is focused on Unix systems.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 2 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Changed.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 5 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: reformulated, please, review it again.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 6 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: done, added some reference about AutotoolsToolchain and PkgConfigDeps

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 12 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:87]: there is no main reference about it, but you can find more points of explanation by using the search bar.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 15 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: added more details about each step.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 19 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: right, but we will postpone windows section for now, too much for this PR.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 20 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 22 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 23 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 25 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 27 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: good catch, fixed, thanks

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 29]
```diff
@@ -0,0 +1,214 @@
...
+
+Please, at first, clone the sources to recreate this project, you can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ in GitHub:
+
+.. code-block:: shell
+
+    git clone https://github.com/conan-io/examples2.git
+    cd examples2/examples/tools/autotools/autotoolstoolchain/string_formatter
+
+We start from a very simple C++ language project with this structure:
```
[the remark this answers is not in the document]
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: fixed

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 28 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: fixed, thanks!

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 10 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 18 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 26 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 14 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Your interior voice 😂 But seriously, there is no problem by using `we`, even the [tutorial](https://docs.conan.io/2/tutorial/consuming_packages/build_simple_cmake_project.html) is using it. The idea is writer be your guide during the steps, like, we are doing it together.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 13 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: improved the line, please, review it again.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 16 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Yes, I improved that line, please, review it again.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst (hunk 21 above)
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst]: Removed Windows section for simplification.

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 30]
```diff
@@ -0,0 +1,178 @@
...
+    AC_CONFIG_SRCDIR([src/main.cpp])
+    AC_CONFIG_FILES([Makefile])
+    PKG_CHECK_MODULES([fmt], [fmt])
+    AC_PROG_CXX
+    AC_OUTPUT
+
+The *Makefile.am* specifies that ``string_formatter`` is the expected executable and that it should be linked to the ``fmt`` library.
+
+.. code-block:: text
+    :caption: **Mafile.am**
```
Comment (memsharded) [inline examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:63]:
```suggestion
    :caption: **Makefile.am**
```

**Review by memsharded — APPROVED** [review]

File: examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst [hunk 31]
```diff
@@ -0,0 +1,178 @@
+.. _examples_tools_autotools_autotools_toolchain_build_project_autotools_toolchain:
+
+Build a simple Autotools project using Conan
+============================================
+
+In this example, we are going to create a string formatter application
+that uses one of the most popular C++ libraries: `fmt <https://fmt.dev/latest/index.html/>`_.
+
+We'll use `Autotools <https://www.gnu.org/software/automake/manual/html_node/Autotools-Introduction.html>`_ as build system and `pkg-config <https://www.freedesktop.org/wiki/Software/pkg-config/>`_ as a helper tool in this case, so you should get them installed
+on Linux and Mac before going forward with this example.
```
[the remark this answers is not in the document]
Comment (uilianries) [reply examples/tools/autotools/autotools_toolchain/build_project_autotools_toolchain.rst:10]: > Are they installed by default in Mac OSX?

No, but they can easily be installed with `brew`, a package manager very popular on OSX.

> Actually those links might not be the best for instructions how to install

Those links are not related how to install, but about the project, in case users want more information.

> Or will be apt install xxxx the best for many users?

I could add section with installation steps on Linux and OSX, but both are not official, as both `brew` and `apt` are not official package managers maintained by Linux or OSX. 

We could explain, "for this example we will use Linux Ubuntu and apt as package manager"

For Mac the same thing, could add `brew` steps.

There are more tools, like `automake` which is not installed by default for both.

**Review by czoido — APPROVED** [review]
